spec/requests/users_controller_spec.rb

```diff
@@ -4080,6 +4080,38 @@ def create_totp
     end
   end
 
+  describe '#disable_second_factor' do
+    context 'when logged in with secure session' do
+      before do
+        sign_in(user)
+        stub_secure_session_confirmed
+      end
+      context 'when user has a registered totp and security key' do
+        before do
+          expect(user.user_second_factors).to be_empty
+          expect(user.security_keys).to be_empty
+
+          totp_second_factor = Fabricate(:user_second_factor_totp, user: user)
+          security_key_second_factor = Fabricate(:user_security_key, user: user, factor_type: UserSecurityKey.factor_types[:second_factor])
+
+          expect(user.reload.user_second_factors.totps.first).to eq(totp_second_factor)
+          expect(user.security_keys.first).to eq(security_key_second_factor)
```

Comment:
```suggestion
```

The use of the fabricator ensures that the object is successfully created so we don't need these assertions too.